watertap/unit_models/thermo_compressor_0D.py

```diff
@@ -0,0 +1,413 @@
+            doc="""Indicates what type of energy balance should be constructed,
+        **default** - EnergyBalanceType.useDefault.
+        **Valid values:** {
+        **EnergyBalanceType.useDefault - refer to property package for default
+        balance type
+        **EnergyBalanceType.none** - exclude energy balances,
+        **EnergyBalanceType.enthalpyTotal** - single enthalpy balance for material,
+        **EnergyBalanceType.enthalpyPhase** - enthalpy balances for each phase,
+        **EnergyBalanceType.energyTotal** - single energy balance for material,
+        **EnergyBalanceType.energyPhase** - energy balances for each phase.}""",
+        ),
+    )
+    CONFIG.declare(
+        "momentum_balance_type",
+        ConfigValue(
+            default=MomentumBalanceType.pressureTotal,
+            domain=In(MomentumBalanceType),
+            description="Momentum balance construction flag",
+            doc="""Indicates what type of momentum balance should be constructed,
+        **default** - MomentumBalanceType.pressureTotal.
+        **Valid values:** {
+        **MomentumBalanceType.none** - exclude momentum balances,
+        **MomentumBalanceType.pressureTotal** - single pressure balance for material,
+        **MomentumBalanceType.pressurePhase** - pressure balances for each phase,
+        **MomentumBalanceType.momentumTotal** - single momentum balance for material,
+        **MomentumBalanceType.momentumPhase** - momentum balances for each phase.}""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        # State blocks
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = True  # motive steam inlet
+        self.properties_motive_steam = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of motive steam inlet",
+            **tmp_dict,
+        )
+
+        tmp_dict["defined_state"] = True  # entrained vapor inlet
+        self.properties_entrained_vapor = (
+            self.config.property_package.state_block_class(
+                self.flowsheet().config.time,
+                doc="Material properties of entrained vapor inlet",
+                **tmp_dict,
+            )
+        )
+
+        tmp_dict["defined_state"] = False
+        self.properties_discharge_mix = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of discharge mix",
+            **tmp_dict,
+        )
+
+        # Ports
+        self.add_port(name="inlet_motive_steam", block=self.properties_motive_steam)
+        self.add_port(
+            name="inlet_entrained_vapor", block=self.properties_entrained_vapor
+        )
+        self.add_port(name="outlet_discharge_mix", block=self.properties_discharge_mix)
```

Comment:
Must our port names be that descriptive? I can see distinguishing between two inlets but any reason we can't just call the outlet something like `outlet`?